solar_system.rb

```diff
@@ -0,0 +1,220 @@
+mars = Planet.new("Mars", 4, "red", -60.0, true, 6_779, 2, 687, 227.9)
+jupiter = Planet.new("Jupiter", 5, "brown-red", 145, false, 139_822, 69, 4380, 778.5)
+hoth = Planet.new("Hoth", 6, "ice-blue", -61.0, true, 116_464, 62, 10_759, 1_429.0)
+# modification from wave 1 to wave 2
+# declaring the planets array remains the same; only difference is that each element in the array goes from a hash in wave 1 to an object of class Planet in wave 2
+planets = [mercury, venus, earth, mars, jupiter, hoth]
+
+# test class Planet planet_info method
+# part of wave 2
+puts hoth.planet_info
+
+# solar system of class SolarSystem created and given an age
+# age part of wave 3 optional
+solar_system_a = SolarSystem.new(28.7, planets)
+
+# test list_planets method
+puts solar_system_a.list_planets
+
+uranus = Planet.new("Uranus", 7, "blue", -216, false, 50_724, 27, 30_770, 2_871)
+
+# test add_planet method
+solar_system_a.add_planet = uranus
+
+# test that list of planets in solar system is updated after adding new planet
+puts solar_system_a.list_planets
+
+# test get_local_year method
+# part of wave 3 optional
+puts solar_system_a.get_local_year(mars)
+
+# test get_distance_between_planets method
+# both outputs below should be the same
+# part of wave 3 optional
+puts solar_system_a.get_distance_between_planets(earth, hoth)
+puts solar_system_a.get_distance_between_planets(hoth, earth)
+
+# ==================
+# user interaction
+# ==================
+
+# display information about any planet
+def display_info(planet_selection)
+  info_about_planet = "#{planet_selection.name} is a(n) #{planet_selection.color} planet. It comes in at \##{planet_selection.order} in the order of planets from the sun. #{planet_selection.name} has a diameter of #{planet_selection.diameter} km and has #{planet_selection.moons} moon(s). It has pleasant weather, averaging about #{planet_selection.temp} degrees Celsius throughout the year. Oh, by the way, one year on #{planet_selection.name} takes about #{planet_selection.year_length} Earth days. That's because it's #{planet_selection.dist_from_sun} million kilometers from the sun."
+  if planet_selection.water
+    info_about_planet += " Luckily, this planet does have water."
+  else
+    info_about_planet += " Unfortunately, this planet lacks water."
+  end
+  return info_about_planet
+end
+
+# ask user to select a planet to explore and allow user to select another
+def planets_info(solar_system)
+  puts "What planet would you like to learn about? Here are your options: "
+
+  solar_system.planets.each_with_index do |planet, planet_order|
+    puts "#{planet_order + 1}. #{planet.name}"
+  end
+
+  # enter the proper name of an existing planet within the solar system you are exploring, e.g., "Mercury", "Jupiter", not "mercury", "jupiter"
+  planet_selection = gets.chomp
+  planet_information = ""
+
+  solar_system.planets.each do |planet|
+    if planet_selection == planet.name
```

Comment:
You're comparing the entry to the Planet's name, but you print out the planet names along with a number.  A user is usually expecting to enter the number to select the planet.  

It's just a little counterintuitive.